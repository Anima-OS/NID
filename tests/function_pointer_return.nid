
// MADNESS return function pointer
//void (*catch_and_return(int abc, char *name_one, char* name_two, int* number))(char* a, char * b, int* c);



// return void(*)(int)
//void (*function_name(int a))(char* arg);

void (*function_name(int a))(char* arg);

void (*(*pointer_to_function_name)(int a))(char * arg);

void* normal_function();

void* (*normal_function_ptr)();