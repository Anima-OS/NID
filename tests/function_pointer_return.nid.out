{ declarations: 
   [ { function: 
        { type: 
           { name: 'void(*)(char*)',
             parameters: 
              [ { type: { name: 'char', pointer: '*' },
                  attributes: {},
                  name: 'arg' } ],
             type: {
               name: 'void'
             },
             attributes: {},
             functionPointer: true },
          attributes: {},
          name: 'function_name',
          parameters: [ { type: { name: 'int' }, name: 'a', attributes: {} } ] } },
     { variable: 
        { type: 
           { name: 'void(*)(char*)',
             parameters: 
              [ { type: { name: 'char', pointer: '*' },
                  attributes: {},
                  name: 'arg' } ],
             type: {
               name: 'void'
             },
             attributes: {},
             functionPointer: true },
          attributes: {},
          name: 'pointer_to_function_name',
          parameters: [ { type: { name: 'int' }, name: 'a', attributes: {} } ],
          functionPointer: true } },
     { function: 
        { type: { name: 'void', pointer: '*' },
          attributes: {},
          name: 'normal_function',
          parameters: [] } },
     { variable: 
        { type: { name: 'void', pointer: '*' },
          attributes: {},
          name: 'normal_function_ptr',
          parameters: [],
          functionPointer: true } } ] }
